cognite/extractorutils/unstable/core/base.py

```diff
@@ -245,6 +256,63 @@ def _setup_logging(self) -> None:
                             sh.setLevel(level_for_handler)
                             root.addHandler(sh)
 
+    def _load_state_store(self) -> None:
+        """
+        Searches through the config object for a StateStoreConfig.
+
+        If found, it will use that configuration to generate a state store, if no such config is found it will either
+        create a LocalStateStore or a NoStateStore depending on whether the ``use_default_state_store`` argument to the
+        constructor was true or false.
+
+        Either way, the state_store attribute is guaranteed to be set after calling this method.
+        """
+
+        def recursive_find_state_store(d: dict[str, Any]) -> StateStoreConfig | None:
+            for k in d:
+                if is_dataclass(d[k]):
+                    res = recursive_find_state_store(d[k].__dict__)
+                    if res:
+                        return res
+                if isinstance(d[k], StateStoreConfig):
+                    return d[k]
+            return None
+
+        state_store_config = recursive_find_state_store(self.application_config.__dict__)
+        if state_store_config:
+            self.state_store = state_store_config.create_state_store(
+                cdf_client=self.cognite_client,
+                default_to_local=self.USE_DEFAULT_STATE_STORE,
+                cancellation_token=self.cancellation_token,
+            )
+        else:
+            self.state_store = (
+                LocalStateStore("states.json", cancellation_token=self.cancellation_token)
+                if self.USE_DEFAULT_STATE_STORE
+                else NoStateStore()
+            )
```

Comment:
I find this easier to read, but just a suggestion
```suggestion
        elif self.USE_DEFAULT_STATE_STORE:
            self.state_store = LocalStateStore("states.json", cancellation_token=self.cancellation_token)
        else:
            self.state_store = NoStateStore()
```

Reply:
makes sense